sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2882,8 +2882,7 @@ object SQLConf {
         "level settings.")
       .version("3.0.0")
       .booleanConf
-      // show full stacktrace in tests but hide in production by default.
-      .createWithDefault(Utils.isTesting)
```

Comment:
Why is this needed?

Reply:
The size of the ErrorInfo is about to exceed the header limit even in tests. So disable this by default.

Reply:
It is disabled by default in python connect tests already: https://github.com/apache/spark/blob/c89221b02bb3000f707a31322e6d40b561e527bd/python/pyspark/testing/connectutils.py#L173

Reply:
Can you rebase and show me it is not an issue anymore.

Reply:
Here is the fix: https://github.com/apache/spark/pull/43017